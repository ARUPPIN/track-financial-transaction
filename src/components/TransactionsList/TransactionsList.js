"use strict";
import React from "react";
import { Spinner } from "reactstrap";
import { TR_LIST_ITEMS } from "../../common/constants";
import "../../styles/TransactionsList.css";

class TransactionsList extends React.Component {
  render() {
    const { transactionList, showSpinner } = this.props;
    return (
      <>
        {transactionList.length > 0 ? (
          <table align="center">
            <thead>
              <tr>
                {TR_LIST_ITEMS.map((headingItem, thIdx) => {
                  return (
                    <th className="th-style" key={thIdx}>
                      {headingItem}
                    </th>
                  );
                })}
              </tr>
            </thead>
            <tbody>
              {transactionList.map((trItem, trIdx) => {
                return (
                  <tr key={trIdx}>
                    {trItem.map((tdItem, tdIdx) => {
                      return (
                        <td className="th-style" key={tdIdx}>
                          {tdItem}
                        </td>
                      );
                    })}
                  </tr>
                );
              })}
            </tbody>
          </table>
        ) : showSpinner ? (
          <div align="center">
            <Spinner color="primary" />
          </div>
        ) : (
          <div align="center">
            No Transactions in specified timeframe! Please input values to
            search for transactions.
          </div>
        )}
      </>
    );
  }
}

export default TransactionsList;
