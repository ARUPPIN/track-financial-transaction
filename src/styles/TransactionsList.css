.transaction-list-view {
  height: 500px;
}
.th-style {
  padding: 10px 30px;
  border: 1px solid black;
}
