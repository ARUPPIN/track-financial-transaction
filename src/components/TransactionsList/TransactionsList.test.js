"use strict";
import React from "react";
import { configure, mount } from "enzyme";
import { mountToJson } from "enzyme-to-json";
import ReactSixteenAdapter from "enzyme-adapter-react-16";
import TransactionsList from "./TransactionsList";
configure({ adapter: new ReactSixteenAdapter() });

describe("TransactionsList", () => {
  const transactionsList = [];
  const component = mount(
    <TransactionsList transactionList={transactionsList} />
  );
  it("snapshot renders", () => {
    let tree = mountToJson(component, { mode: "deep" });
    expect(tree).toMatchSnapshot();
  });
});
