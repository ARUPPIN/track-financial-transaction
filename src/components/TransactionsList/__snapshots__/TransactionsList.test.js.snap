// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsList snapshot renders 1`] = `
<div
  align="center"
>
  No Transactions in specified timeframe! Please input values to search for transactions.
</div>
`;
